### PR TITLE
feat(tasks): unassign a task

### PR DESCRIPTION
**Pull request by cexbrayat**
A little contribution, to understand a bit better how it works.
👍 design, really easy to add a feature.

### CODE REVIEW

File: backend/src/main/java/org/globe42/web/tasks/TaskController.java [hunk 1]
```diff
@@ -92,10 +92,13 @@ public TaskController(TaskDao taskDao,
     @ResponseStatus(HttpStatus.CREATED)
     public void assign(@PathVariable("taskId") Long taskId, @Validated @RequestBody TaskAssignmentCommandDTO command) {
```
Comment (jnizet) [inline backend/src/main/java/org/globe42/web/tasks/TaskController.java]: I'd rather leave the NotNull annotation in TaskAssignmentCommandDTO there, and use a separate, `DeleteMapping("/{taskId}/assignments")` method to delete the assignment. That keeps validation there, keeps the semantics of the command, and looks more RESTful to me.

File: frontend/src/app/tasks-page/tasks-page.component.ts [hunk 2]
```diff
@@ -59,6 +62,8 @@ export class TasksPageComponent implements OnInit {
     action.switchMap(() => this.tasksResolverService.resolve(this.route.snapshot))
       .subscribe(
         page => {
+          // maybe we are displaying a page that doesn't exist anymore
```
Comment (jnizet) [inline frontend/src/app/tasks-page/tasks-page.component.ts]: 👍

File: frontend/src/app/tasks/tasks.component.html [hunk 3]
```diff
@@ -15,7 +15,8 @@
       </div>
       <div class="task-buttons" *ngIf="task.model.status === 'TODO'">
         <button (click)="edit(task, $event)" class="btn btn-secondary btn-sm edit-button"><span class="fa fa-edit"></span></button>
-        <button (click)="assign(task, $event)" class="btn btn-outline-primary btn-sm assign-button" ><span class="fa fa-hand-o-right"></span></button>
+        <button (click)="assign(task, $event)" class="btn btn-outline-primary btn-sm assign-button" *ngIf="!task.model.assignee"><span class="fa fa-hand-o-right"></span></button>
+        <button (click)="unassign(task, $event)" class="btn btn-outline-secondary btn-sm unassign-button" *ngIf="task.model.assignee"><span class="fa fa-hand-o-left"></span></button>
```
Comment (jnizet) [inline frontend/src/app/tasks/tasks.component.html]: I'd rather use this icon: http://fontawesome.io/icon/hand-paper-o/: it's more different than the one used to assign, and can be seen as a "no, stop, I don't want to do this" sign.

File: backend/src/main/java/org/globe42/web/tasks/TaskController.java (hunk 1 above)
Comment (cexbrayat) [inline backend/src/main/java/org/globe42/web/tasks/TaskController.java]: Ha! I hesitated :) Now done in the amended commit.

File: backend/src/main/java/org/globe42/web/tasks/TaskController.java [hunk 4]
```diff
@@ -98,6 +91,13 @@ public void assign(@PathVariable("taskId") Long taskId, @Validated @RequestBody
         task.setAssignee(user);
     }
 
+    @DeleteMapping("/{taskId}/assignments")
+    @ResponseStatus(HttpStatus.CREATED)
```
Comment (jnizet) [inline backend/src/main/java/org/globe42/web/tasks/TaskController.java]: NO_CONTENT should be used.

Comment (cexbrayat) [inline backend/src/main/java/org/globe42/web/tasks/TaskController.java]: Right, fixed

**Pull request merged**